test/onion-sticky.spec.js

```diff
@@ -0,0 +1,370 @@
+    it('true when scrolled above bottom point', function () {
+      document.body.scrollTop = this.container.height() + this.header.height() + 10;
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.containerBottomAboveBottomPoint());
+    });
+
+    it('false when scrolled below bottom point', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.containerBottomAboveBottomPoint());
+    });
+  });
+
+  describe('elementBottomAboveBottomPoint', function () {
+    it('true when scrolled above bottom point', function () {
+      document.body.scrollTop = this.element.height() + this.header.height() + 10;
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.elementBottomAboveBottomPoint());
+    });
+
+    it('false when scrolled below bottom point', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.elementBottomAboveBottomPoint());
+    });
+  });
+
+  describe('elementTopBelowZeroPoint', function () {
+    it('true when scrolled beyond zero point', function () {
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.elementTopBelowZeroPoint());
+    });
+
+    it('false when not scrolled beyond zero point', function () {
+      this.element.css({ position: 'relative', top: '100px' });
+      document.body.scrollTop = this.header.height();
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.elementTopBelowZeroPoint());
+    });
+  });
+
+  describe('containerTopBelowZeroPoint', function () {
+    it('true when container top below zero point', function () {
+      this.container.css({ position: 'relative', top: '1px' })
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.containerTopBelowZeroPoint());
+    });
+
+    it('false when cointer top above zero point', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.containerTopBelowZeroPoint());
+    });
+  });
+
+  describe('elementFixedToTop', function () {
+    it('true when element fixed to top', function () {
+      this.element.css({ top: '10px', position: 'fixed' });
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.elementFixedToTop());
+    });
+
+    it('false when element not fixed to top', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.elementFixedToTop());
+    });
+  });
+
+  describe('elementFixedToBottom', function () {
+    it('true when element fixed to bottom', function () {
+      this.element.css({ bottom: '10px', position: 'fixed' });
+      this.sticky.getFrameData();
+
+      assert.isTrue(this.sticky.elementFixedToBottom());
+    });
+
+    it('false when element not fixed to bottom', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.elementFixedToBottom());
+    });
+  });
+
+  describe('calculateScrollDirection', function () {
+    it('SCROLLING_UNCH on first call', function () {
```

Comment:
I have been infected by watching stock market reports as a child. changing to: `UNCHANGED`